Source/SwiftLintFramework/Helpers/library_wrapper.swift

```diff
@@ -0,0 +1,195 @@
+        /// If we hit this path, the user is invoking Swift via swiftenv shims and has not set the
+        /// environment variable; this means we're going to end up trying to load from `/usr/lib`
+        /// which will fail - and instead, we can give a more useful error message.
+        fatalError("Swift is installed via swiftenv but swiftenv is not initialized.")
+    }
+
+    if !swiftPath.hasSuffix("/bin/swift") {
+        return nil
+    }
+
+    /// .../bin/swift -> .../lib
+    return swiftPath.deleting(lastPathComponents: 2).appending(pathComponent: "/lib")
+}()
+
+/// Fallback path on Linux if no better option is available.
+internal let linuxDefaultLibPath = "/usr/lib"
+
+let toolchainLoader = Loader(searchPaths: [
+    linuxSourceKitLibPath,
+    linuxFindSwiftenvActiveLibPath,
+    linuxFindSwiftInstallationLibPath,
+    linuxDefaultLibPath
+].compactMap({ $0 }))
+
+#else
+
+// MARK: - Darwin
+
+let toolchainLoader = Loader(searchPaths: [
+    xcodeDefaultToolchainOverride,
+    toolchainDir,
+    xcrunFindPath,
+    /*
+    These search paths are used when `xcode-select -p` points to
+    "Command Line Tools OS X for Xcode", but Xcode.app exists.
+    */
+    applicationsDir?.xcodeDeveloperDir.toolchainDir,
+    applicationsDir?.xcodeBetaDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeDeveloperDir.toolchainDir,
+    userApplicationsDir?.xcodeBetaDeveloperDir.toolchainDir
+].compactMap { path in
+    if let fullPath = path?.usrLibDir, FileManager.default.fileExists(atPath: fullPath) {
+        return fullPath
+    }
+    return nil
+})
+
+/// Returns "XCODE_DEFAULT_TOOLCHAIN_OVERRIDE" environment variable
+///
+/// `launch-with-toolchain` sets the toolchain path to the
+/// "XCODE_DEFAULT_TOOLCHAIN_OVERRIDE" environment variable.
+private let xcodeDefaultToolchainOverride = env("XCODE_DEFAULT_TOOLCHAIN_OVERRIDE")
+
+/// Returns "TOOLCHAIN_DIR" environment variable
+///
+/// `Xcode`/`xcodebuild` sets the toolchain path to the
+/// "TOOLCHAIN_DIR" environment variable.
+private let toolchainDir = env("TOOLCHAIN_DIR")
+
+/// Returns toolchain directory that parsed from result of `xcrun -find swift`
+///
+/// This is affected by "DEVELOPER_DIR", "TOOLCHAINS" environment variables.
+private let xcrunFindPath: String? = {
+    let pathOfXcrun = "/usr/bin/xcrun"
+
+    if !FileManager.default.isExecutableFile(atPath: pathOfXcrun) {
+        return nil
+    }
+
+    guard let output = Exec.run(pathOfXcrun, "-find", "swift").string else {
+        return nil
+    }
+
+    var start = output.startIndex
+    var end = output.startIndex
+    var contentsEnd = output.startIndex
+    output.getLineStart(&start, end: &end, contentsEnd: &contentsEnd, for: start..<start)
+    let xcrunFindSwiftPath = String(output[start..<contentsEnd])
+    guard xcrunFindSwiftPath.hasSuffix("/usr/bin/swift") else {
+        return nil
+    }
+    let xcrunFindPath = xcrunFindSwiftPath.appending(pathComponent: "macosx")
```

Comment:
I had to change this from the original `library_wrapper` from SourceKitten ⚠️

Do I need to change something in the Linux code path?

Reply:
This line should be same with original like:
```suggestion
    let xcrunFindPath = xcrunFindSwiftPath.deleting(lastPathComponents: 3)
```